submarine-test/test-k8s/src/test/java/org/apache/submarine/server/client/JobRestApiIT.java

```diff
@@ -0,0 +1,182 @@
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.apache.commons.httpclient.methods.DeleteMethod;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.io.FileUtils;
+import org.apache.submarine.server.api.job.Job;
+import org.apache.submarine.server.api.job.JobId;
+import org.apache.submarine.server.api.job.JobServiceException;
+import org.apache.submarine.server.json.JobIdDeserializer;
+import org.apache.submarine.server.json.JobIdSerializer;
+import org.apache.submarine.server.response.JsonResponse;
+import org.apache.submarine.server.rest.RestConstants;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@SuppressWarnings("rawtypes")
+public class JobRestApiIT {
+  private static final Logger LOG = LoggerFactory.getLogger(JobRestApiIT.class);
+
+  private static String JOB_PATH = "/api/" + RestConstants.V1 + "/" + RestConstants.JOBS;
+
+  private Gson gson = new GsonBuilder()
+      .registerTypeAdapter(JobId.class, new JobIdSerializer())
+      .registerTypeAdapter(JobId.class, new JobIdDeserializer())
+      .create();
+
+  @Test
+  public void testJobServerPing() throws IOException {
+    GetMethod response = HttpClientManager.doGet("/api/" + RestConstants.V1 + "/"
+        + RestConstants.JOBS + "/" + RestConstants.PING);
+    String requestBody = response.getResponseBodyAsString();
+    Gson gson = new Gson();
+    JsonResponse jsonResponse = gson.fromJson(requestBody, JsonResponse.class);
+    Assert.assertEquals(Response.Status.OK.getStatusCode(), jsonResponse.getCode());
+    Assert.assertEquals("Pong", jsonResponse.getResult().toString());
+  }
+
+  @Test
+  public void testTensorFlowWithJsonSpec() throws Exception {
+    String body = loadContent("tensorflow/tf-mnist-req.json");
+    String patchBody = loadContent("tensorflow/tf-mnist-patch-req.json");
+    run(body, patchBody, "application/json");
+  }
+
+  @Test
+  public void testTensorFlowWithYamlSpec() throws Exception {
+    String body = loadContent("tensorflow/tf-mnist-req.yaml");
+    String patchBody = loadContent("tensorflow/tf-mnist-patch-req.yaml");
+    run(body, patchBody, "application/yaml");
+  }
+
+  @Test
+  public void testPyTorchWithJsonSpec() throws Exception {
+    String body = loadContent("pytorch/pt-mnist-req.json");
+    String patchBody = loadContent("pytorch/pt-mnist-patch-req.json");
+    run(body, patchBody, "application/json");
+  }
+
+  @Test
+  public void testPyTorchWithYamlSpec() throws Exception {
+    String body = loadContent("pytorch/pt-mnist-req.yaml");
+    String patchBody = loadContent("pytorch/pt-mnist-patch-req.yaml");
+    run(body, patchBody, "application/yaml");
+  }
+
+  private void run(String body, String patchBody, String contentType) throws Exception {
```

Comment:
We need to directly obtain the job information in k8s through kubectl api in the `run` function for verification `create job\get job\delete job`.